app/controllers/application_controller.rb

```diff
@@ -8,7 +8,7 @@ def cart
   end
 
   def current_user
-    User.find(session[:user_id])
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
```

Comment:
Refactored this method to use memoization. Uses variable on the left if it exists, and if it does not exist, does the operation on the right. Prevents from having to go to the db multiple times if we need to check current user more than once in a single request.

Reply:
Great refactor; I can see how this is less expensive now - thanks for the detailed explanation

Reply:
Great refactor; I can see how this is less expensive now - thanks for the detailed explanation